src/app/copy/block-copy/block-copy.component.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  ViewEncapsulation
+} from '@angular/core';
+
+import { CopyService } from '../copy-service/copy.service';
+
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'pfng-block-copy',
+  templateUrl: './block-copy.component.html',
+  styleUrls: ['./block-copy.component.less']
+})
+
+export class BlockCopyComponent implements OnInit {
+  @Input('label') label: string;
+  @Input('copyBtnAriaLabel') copyBtnAriaLabel: string;
+  @Input('expandBtnAriaLabel') expandBtnAriaLabel: string;
+  @Input('tooltip') tooltip: string;
+  @Input('copyValue') copyValue: string = 'Missing \'copyValue\' @Input property';
+  @Input('buttonLabel') buttonLabel: string = 'Copy';
+  @Input('expanded') expanded: boolean = false;
+
+  @Output('copiedToClipboard') copiedToClipboard = new EventEmitter();
+
+  /**
+   * Used to uniquly relate label to copy button
+   */
+  public _hash: number = Math.floor(Math.random() * 10000);
+
+  get hash(): number {
+    return this._hash;
+  }
+
+  /**
+   * The default constructor
+   */
+  constructor(private copyService: CopyService) {}
+
+  /**
+   * Toggle copyValue panel open and close
+   */
+  togglePanel(): void {
+    this.expanded = !this.expanded;
+  }
+
+  /**
+   * Copy copyValue to the user's system clipboard
+   */
+  copyValueToClipboard(): void {
+      let result = this.copyService.copy(this.copyValue);
+      if (result) {
+        this.copiedToClipboard.emit(`${this.label} copied!`);
+      } else {
+        console.error(`Failed to copy ${this.label}`);
+      }
+  }
+
+  ngOnInit() {
+    if (!this.label) throw new Error('Missing required @Input property \'label\'');
```

Comment:
Good suggestions - I'm with ya. It's sort of where I was headed initially when I was requiring the label, and using that to construct unique aria label defaults. For screen readers, it's important for the aria labels to be unique per component because if they are not, and there are more than one in a single page, it's impossible to tell the copy/expand buttons apart from one another.

I'm starting to think it's a problem we shouldn't try to solve for people at the component level, and instead just leave the component in a most flexible state. It doesn't ensure the component will be "accessible" in all scenarios, but it may be a reservation we have to make.

I'll keep thinking about this, in the mean time I'll go ahead and push my next set of updates.